### ISSUE TITLE
Artists are a list of chars and not a string

### ISSUE THREAD

**Issue by Nicolas1847** [krateng/maloja#179]
Hi

First of all, thank you for your work on maloja. I love to see that kind of self hostable service and using those.

I'm using maloja for the first time as a docker container, stats are send from funkwhale. The plugin configuration from Funkwhale is ok, listened tracks are send to Maloja without issue, but the informations for the track artists are not handled (or send?) correctly.

As you can see, the track_artists is handled as a list of chars and not a string. 

```
[database] Incoming scrobble [Client: default | API: native/v1]: {'track_artists': ['V', 'a', 'r', 'i', 'o', 'u', 's', ' ', 'A', 'r', 't', 'i', 's', 't', 's'], 'track_title': 'Crystal Atlantic - Nihon Nights', 'album_name': 'Faraway Beach - [Future Funk Mix]', 'scrobble_time': 1670518802}
[dbcache] Invalidated 95 of 535 DB cache entries
[dbcache] Cache: Size: 440 | Hits: 1528/2063 | Estimated Memory: 22 KiB
[dbcache] Entity Cache: Size: 13 | Hits: 64/77 | Estimated Memory: 1 KiB
[dbcache] System RAM Utilization: 28.3%
[sqldb] Database Cleanup...
[sqldb] Database Cleanup complete!
[database] Incoming scrobble [Client: default | API: native/v1]: {'track_artists': ['S', 'h', 'a', 'r', 'o', 'n', ' ', 'J', 'o', 'n', 'e', 's', ' ', '&', ' ', 'T', 'h', 'e', ' ', 'D', 'a', 'p', '-', 'K', 'i', 'n', 'g', 's'], 'track_title': 'What If We All Stopped Paying Taxes?', 'album_name': 'Soul Time!', 'scrobble_time': 1670519023}
[dbcache] Invalidated 18 of 458 DB cache entries
[database] Incoming scrobble [Client: default | API: native/v1]: {'track_artists': ['U', 'n', 'd', 'e', 'r', 'w', 'o', 'r', 'l', 'd'], 'track_title': 'Moaner', 'album_name': 'Beaucoup Fish', 'scrobble_time': 1670519436}
[dbcache] Invalidated 9 of 449 DB cache entries
```

By doing so, the stats are totally irrelevant, as I have only artists represented by a letter and nothing else

![image](https://user-images.githubusercontent.com/6197408/206519929-952b8120-f6d3-4fa7-b758-8a5eeced16cf.png)

Any idea of what is happening here?

**Comment by Nicolas1847** [comment]
Ok, so I've set up a new local funkwhale instance on a test VM and tried different stuff. 
Maloja and Funkwhale are still on the same server.

First of all, the docker version and pip version both have the same behaviour.

But I tried also to sniff the traffic between the two to see if I could see something different.

I've attached a pcap file (that you can open with tools like wireshark) to see what is transmitted from FW and check if something is incorrect.

If I understand correctly, on packet 1817, the datas are send to Maloja through post on the api and we can see that the informations are send normally.

On packet 1822, Maloja returns 200 with a JSON that resumes what was send, and we can see that the artist is now splitted as an array of chars.

[dump.pcap.zip](https://github.com/krateng/maloja/files/10240614/dump.pcap.zip)


I hope this will help.

**Comment by Nicolas1847** [comment]
Nothing here? :(

**Comment by phw** [comment]
@Nicolas1847 I wrote the Funkwhale plugin and I experience the same issue. Started looking into this.

So far I am not entirely sure what's happening. One thing is that the plugin needs some updates, as Maloja changed some details on how the submission works. But even with the necessary changes I get the same problem, while submitting the very same payload via a manual HTTP call outside the plugin works.

My current assumption is that it must be some issue on how the plugin works, but I have not yet fully debugged it. If it is an issue on the Funkwhale side I'll see to get the plugin fixed before the FW 1.3 final release.

**Comment by phw** [comment]
Got it fixed, first properly submitted artist again:

![grafik](https://user-images.githubusercontent.com/29852/213790710-8e20f021-f205-4b5e-afc1-c77aab7d28b7.png)

Fix is at https://dev.funkwhale.audio/funkwhale/funkwhale/-/merge_requests/2328 . I think you can close the issue here then.

**Comment by Nicolas1847** [comment]
thank you for your work ! 

I'm closing the issue

**Comment by phw** [comment]
Just to have this information here: The fix for the Funkwhale plugin has been merged and will be part of the final FW 1.3 release. 

The reason for this being broken is twofold: Maloja changed the content of the `artists`  field in the submission to be a list instead of a single string, and the plugin was still following the old approach. Secondly the plugin did not submit the data as JSON but form encoded. Obviously Maloja was able to deceiver both as long as it was only plain strings, but there is no clear way to encode multiple values of a list in form encoded data.
